pkg/client/routegenericinformers/interface.go

```diff
@@ -0,0 +1,80 @@
+package genericinformers
+
+import (
+	"k8s.io/klog/v2"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/informers"
+)
+
+type GenericResourceInformer interface {
+	ForResource(resource schema.GroupVersionResource) (informers.GenericInformer, error)
+	Start(stopCh <-chan struct{})
+}
+
+// GenericInternalResourceInformerFunc will return an internal informer for any resource matching
+// its group resource, instead of the external version. Only valid for use where the type is accessed
+// via generic interfaces, such as the garbage collector with ObjectMeta.
+type GenericInternalResourceInformerFunc func(resource schema.GroupVersionResource) (informers.GenericInformer, error)
```

Comment:
which part of the route controller uses this?

Reply:
nice catch. As far as I can tell, it is a dead code and is not used. This applies to the openshift-controller-manager as well. 

- I can remove this whole file
- I can do a followup and remove it from the main controller as well

Reply:
The generic informer itself is not used by the route controllers.  The route controllers use specific informers.

Reply:
> This applies to the openshift-controller-manager as well.

> I can do a followup and remove it from the main controller as well

I was referring to the above comments about the main controller. I will wait for the changes to see the final outcome :)

Reply:
all of the code related to generic informers will be removed, even the `ControllerContext.ToGenericInformer()` invocation